doc/rfcs/003-quickly-collaborate-with-coworkers-and-friends.md

```diff
@@ -0,0 +1,118 @@
+
+### Streamline collaboration while avoiding harassment
+
+With the approach described above, we believe Teletype can provide a streamlined set-up process while also preventing harassment. In order for Teletype to prompt you to join someone's portal, you must first have collaborated with that person in the past (by sending your portal URL to them via a third-party service or by joining their portal via a URL they sent you via a third-party service) *and* that person must still exist in your list of past collaborators (i.e., you haven't removed them from your list of past collaborators). In other words, people that you've never collaborated with cannot cause portal invitations to appear in your editor. And people that you've collaborated with in the past, but don't want to receive invitations from in the future, cannot cause portal invitations to appear in your editor either.
+
+## Drawbacks
+
+Why should we *not* do this?
+
+*TODO*
+
+## Rationale and alternatives
+
+- Why is this approach the best in the space of possible approaches?
+    - *TODO*
+- What other approaches have been considered and what is the rationale for not choosing them?
+    - *TODO*
+- What is the impact of not doing this?
+    - *TODO*
+
+## Unresolved questions
+
+- What unresolved questions do you expect to resolve through the RFC process before this gets merged?
+    - If I invite a past collaborator to join my portal, and that person has multiple Atom windows open, can we show the invitation only in the frontmost window?
+    - If I invite a past collaborator to join my portal, and that person has Atom windows open on multiple computers, are we OK with showing the invitation on each computer?
+    - *TODO*: What else?
+- What unresolved questions do you expect to resolve through the implementation of this feature before it is released in a new version of the package?
+    - *TODO*
+- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?
+    - *TODO*
+
+## Even more TODOs 😇
+
+- Explicitly state that past collaborators have two possible states: online or offline. Setting your status to "away" or "busy" is out of scope for now. Making yourself invisible is out of scope for now. In the meantime, you sign out of Teletype or disable Teletype in Atom's package settings when you want to use Atom while not accepting invitations from your past collaborators.
+- What happens if I'm currently a guest and I click to invite someone to a portal?
+- How should we handle an extremely large list of past collaborators? How should we sort it? Should the user be able to change how it's sorted? Should they be able to filter the list somehow?
```

Comment:
- Sort by how recently you collaborated with someone.
- Automatically remove people if you haven't collaborated for a long time. Like 2-3 months. This might feel intrusive, but at the same time it let's you keep inviting more people without having to worry about cleaning up your list. Your list stays current automatically. And you can always invite people again.
- Then the list stops at like 50 people? With a filter input at the top to quickly find someone.
- The filter input only shows if you have more than 10 people in your list, otherwise it feels unnecessary.

Reply:
Thanks, @simurai. I :heart: these ideas. 38a3b71875d1ab4e14ef1c1852cabbead343791f proposes an MVP for the initial implementation, but your notes above feel like a solid direction to move in once we have the barebones functionality in place.